File: 118.py
class Solution(object):
    def generate(self, n):
        """
        :type numRows: int
        :rtype: List[List[int]]
        """
        if n <= 0:return []
        res = [[1]]
        for _ in range(n-1):
            temp = [0] + res[-1] +[0]
            res.append([temp[i]+temp[i+1] for i in range(len(temp)-1)])
        return res

File: 118.cpp
class Solution {
public:
    vector<vector<int>> generate(int numRows) {
        vector<vector<int>> np;
        np.resize(numRows);
        for(int i = 0; i < numRows; i++){
            np[i].resize(i+1);
            np[i][0] = 1;
            for ( int col = 1; col <= i; col++){
                np[i][col] = np[i-1][col-1] + (col < i? np[i-1][col]:0);
            }
        }
        return np;
    }
};
